core/server/data/migrations/versions/3.22/05-update-type-for-settings.js

```diff
@@ -0,0 +1,121 @@
+const logging = require('../../../../../shared/logging');
+
+// type mapping for settings. object types are ignored for now
+const typeMapping = [
+    {
+        keys: [
+            'db_hash',
+            'session_secret',
+            'theme_session_secret',
+            'ghost_public_key',
+            'ghost_private_key',
+            'title',
+            'description',
+            'logo',
+            'cover_image',
+            'icon',
+            'accent_color',
+            'lang',
+            'timezone',
+            'codeinjection_head',
+            'codeinjection_foot',
+            'facebook',
+            'twitter',
+            'meta_title',
+            'meta_description',
+            'og_image',
+            'og_title',
+            'og_description',
+            'twitter_image',
+            'twitter_title',
+            'twitter_description',
+            'active_theme',
+            'password',
+            'public_hash',
+            'members_public_key',
+            'members_private_key',
+            'members_email_auth_secret',
+            'default_content_visibility'
+        ],
+        type: 'string'
+    },
+    {
+        keys: [
+            'notifications',
+            'navigation',
+            'secondary_navigation',
+            'slack',
+            'shared_views',
+            'portal_plans'
+        ],
+        type: 'array'
+    },
+    {
+        keys: [
+            'next_update_check'
+        ],
+        type: 'number'
+    },
+    {
+        keys: [
+            'amp',
+            'is_private',
+            'portal_name',
+            'portal_button'
+        ],
+        type: 'boolean'
+    },
+    {
+        keys: [
+            'labs',
+            'unsplash',
+            'members_subscription_settings',
+            'stripe_connect_integration',
+            'bulk_email_settings'
+        ],
+        type: 'object'
+    }
+];
+
+module.exports = {
+    config: {
+        transaction: true
+    },
+
+    async up(options) {
+        // set the new type for each setting
+        await Promise.map(typeMapping, async (typeMap) => {
+            return await Promise.map(typeMap.keys, async (key) => {
```

Comment:
One thing I was thinking about is if we should be a little more careful and add `{concurrency: 100}` to these nested maps like we did [here](https://github.com/TryGhost/Ghost/blob/91e3630f3647e6bdfc24eba8b2ec4c1d29866ab7/core/server/data/migrations/versions/1.25/1-update-koenig-beta-html.js#L64). But I think having only `53` properties to migrate it should not be an issue :thinking:

Reply:
🤔 Hmm yeah I _think_ it should be fine in that case but good point. Will leave as is for now but keep in mind when we do more testing at the end 👍